types/object.go

```diff
@@ -55,6 +55,12 @@ func (o ObjectMode) String() string {
 	return strings.Join(s, "|")
 }
 
+func (o ObjectMode) Add(mode ObjectMode) {
+	o |= mode
+}
+func (o ObjectMode) Del(mode ObjectMode) {
```

Comment:
Please add unit tests and example tests in `object_test.go`

```go
func TestObjectMode_Add(t *testing.T) {
    cases := ...
    for _, v := range cases {
          t.Run(xxxx)
    }
}

func ExampleObjectMode_Add() {
    var o ObjectMode
    o.Add(ModeDir)
    o.Add(ModeRead | ModeAppend)
}
```

Reply:
In order to check value at local, we may need to change `(o ObjectMode) Del` to `(o *ObjectMode) Del`

Reply:
We may need to change `o ObjectMode` into `o *ObjectMode`